inc/class-s3-media-sync.php

```diff
@@ -21,7 +21,11 @@ class S3_Media_Sync {
 	public function __construct( S3_Media_Sync_Settings $settings_handler ) {
 		$this->settings_handler = $settings_handler;
 		$this->settings        = $this->settings_handler->get_settings();
-		$this->bucket         = S3_Bucket::from_settings($this->settings);
+		
+		// Only create bucket if we have required settings
```

Comment:
Nit: you can avoid storing `$this->settings` just to build `$this->bucket`. Fetching via `$this->settings_handler->get_settings()` on demand keeps a single source of truth and simplifies constructor and tests.